File: src/main/java/com/leijendary/spring/microservicetemplate/json/deserializer/StringTrimDeserializer.java
package com.leijendary.spring.microservicetemplate.json.deserializer;

import com.fasterxml.jackson.core.JsonParser;
import com.fasterxml.jackson.databind.DeserializationContext;
import com.fasterxml.jackson.databind.deser.std.StringDeserializer;
import org.springframework.boot.jackson.JsonComponent;

import java.io.IOException;

import static java.util.Optional.ofNullable;

@JsonComponent
public class StringTrimDeserializer extends StringDeserializer {

    @Override
    public String deserialize(final JsonParser jsonParser,
                              final DeserializationContext deserializationContext) throws IOException {
        // Get the result from the original StringDeserializer
        final var deserialized = super.deserialize(jsonParser, deserializationContext);

        // ... and just trim it
        return ofNullable(deserialized)
                .map(String::trim)
                .orElse(deserialized);
    }
}

File: src/main/kotlin/com/leijendary/spring/template/core/json/StringTrimDeserializer.kt
package com.leijendary.spring.template.core.json

import com.fasterxml.jackson.core.JsonParser
import com.fasterxml.jackson.databind.DeserializationContext
import com.fasterxml.jackson.databind.deser.std.StringDeserializer
import org.springframework.boot.jackson.JsonComponent

private val exclusions = arrayOf("currentPassword", "password")

@JsonComponent
class StringTrimDeserializer : StringDeserializer() {
    override fun deserialize(jsonParser: JsonParser, deserializationContext: DeserializationContext): String? {
        // Get the result from the original StringDeserializer
        val deserialized = super.deserialize(jsonParser, deserializationContext)

        // But skip the whitelisted fields
        if (exclusions.contains(jsonParser.currentName())) {
            return deserialized
        }

        // ... and just trim it
        return deserialized?.let { obj -> obj.trim { it <= ' ' } }
    }
}
